### ISSUE TITLE
Make automation of some basic processing parameters by key frames.

### ISSUE THREAD

**Issue by bouncyball-git** [ilia3101/MLV-App#119]
I thought about this long while ago. What if make in time automation of some basic processing parameters like exposure, light/dark range (maybe gradient stuff also) by interpolating value between keyframes (over actual frames).

I know it won't be easy but we could try :)

**Comment by masc4ii** [comment]
I also thought about that. But this is indeed a monster task. First we need a kind of keyframe editor, then we need a data structure to save all those keyframes. Keyframes maybe different for each parameter... and so on...
One little thing could make it a little easier: there is a Qt Automation class, which does linear interpolation between 2 values on one parameter. But I think, this is the easiest thing in this task.

**Comment by bouncyball-git** [comment]
Yea, we could start with one parameter without key frame editor only with simple structure for now to test.

**Comment by masc4ii** [comment]
It will also dramatically drop framerate. Because all matrices will be calculated for each frame within the animation.
In ReceiptSettings.cpp all members should be QVector<....> and all public functions should get a frame parameter.
Oh man... but then we should get the same problem as with the "Save on Quit" thing: if the code changes a parameter, it is the same as the user changes it (with the current strategy) -> nobody nows the reason for a parameter change. We would have to invent a completely new parameter input and update strategy... sh**.

**Comment by bouncyball-git** [comment]
Yeah... u r absolutely right. At least we could do key frame processing optional during playback.

**Comment by bouncyball-git** [comment]
QVector type array also quite slow right?

**Comment by masc4ii** [comment]
I really think this would be MLVApp 2.0 - This should not really be possible with current archtecture and strategy. When I started I did not think about something like this could ever happen 😄 

Really better would be here to start at zero - with all our knowledge we got in the last months.

QVector should not be slower then something else - I don't expect a problem there.

**Comment by bouncyball-git** [comment]
> and someone who really knows creating good software objects architecture...

Think it's all said already. We so many times told people on forum to join us but, alas...
If someone wants job to be done he/she should do it him/herself ;)

**Comment by ilia3101** [comment]
I remember dreaming of this about a year ago. Maybe for 2.0 then ;)

**Comment by bouncyball-git** [comment]
Nice to hear that we all have the same vision and desires :grinning:
